projects/components/src/lib/directives/show-clipped-text.directive.ts

```diff
@@ -0,0 +1,278 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { Directive, ElementRef, Input, OnDestroy, OnInit } from '@angular/core';
+
+export enum TooltipPosition {
+    tl = 'tooltip-top-left',
+    tr = 'tooltip-top-right',
+    br = 'tooltip-bottom-right',
+    bl = 'tooltip-bottom-left',
+}
+
+export enum TooltipSize {
+    sm = '100px',
+    md = '200px',
+    lg = '400px',
+}
+
+/**
+ * Singleton tooltip created by directive
+ */
+const tip = {
+    /** A single DOM node structure for the popup is created and shared with all instances (the .tooltip)  */
+    container: null as HTMLElement,
+
+    /** The inner node (the .tooltip-content) */
+    content: null as HTMLElement,
+
+    /** Timer for setTimeout used when hiding */
+    hideTimeout: null as number,
+
+    /** The directive that last caused the tooltip to be displayed */
+    currentDirective: null as ShowClippedTextDirective,
+
+    /** Whether the mouse is currently over a host or the tooltip */
+    isMouseOver: false,
+
+    /**
+     * The  that last caused the
+     */
+    get currentHost(): HTMLElement {
+        return tip.currentDirective.hostElement;
+    },
+
+    get tooltipSize(): TooltipSize {
```

Comment:
Fix top comment (get currentHost), and add comment to below getter (get tooltipSize)?